packages/dd-trace/src/ci-visibility/intelligent-test-runner/get-itr-configuration.js

```diff
@@ -15,28 +15,31 @@ function getItrConfiguration ({
   runtimeVersion,
   branch
 }, done) {
-  const intakeUrl = url || new URL(`https://api.${site}`)
-
-  const apiKey = process.env.DATADOG_API_KEY || process.env.DD_API_KEY
-  const appKey = process.env.DATADOG_APP_KEY ||
-    process.env.DD_APP_KEY ||
-    process.env.DATADOG_APPLICATION_KEY ||
-    process.env.DD_APPLICATION_KEY
-
-  if (!apiKey || !appKey) {
-    done(new Error('App key or API key undefined'))
-    return
-  }
-
   const options = {
     path: '/api/v2/libraries/tests/services/setting',
     method: 'POST',
     headers: {
-      'dd-api-key': apiKey,
-      'dd-application-key': appKey,
       'Content-Type': 'application/json'
     },
-    url: intakeUrl
+    url
+  }
+
+  if (isEvpProxy) {
+    options.path = '/evp_proxy/v2/api/v2/libraries/tests/services/setting'
+    options.headers['X-Datadog-EVP-Subdomain'] = 'api'
+    options.headers['X-Datadog-NeedsAppKey'] = 'true'
+  } else {
+    const apiKey = process.env.DATADOG_API_KEY || process.env.DD_API_KEY
+    const appKey = process.env.DATADOG_APP_KEY ||
+      process.env.DD_APP_KEY ||
+      process.env.DATADOG_APPLICATION_KEY ||
+      process.env.DD_APPLICATION_KEY
+
+    if (!apiKey || !appKey) {
+      return done({ err: new Error('App key or API key undefined') })
```

Comment:
Not very Node.js-like putting the error in an object property rather than splitting the object to a separate argument. 🤔

Reply:
true. Fixed in https://github.com/DataDog/dd-trace-js/pull/2584/commits/b7ffb778d8e282067ea59ba73348099523d68d9c